test/integration/src/main/java/io/pravega/test/integration/compatibility/CompatibilityChecker.java

```diff
@@ -0,0 +1,215 @@
+        readerGroupManager.createReaderGroup(readerGroupId, readerGroupConfig);
+        ReaderGroup readerGroup = readerGroupManager.getReaderGroup(readerGroupId);
+
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName,new UTF8StringSerializer(), EventWriterConfig.builder().build());
+        for( int event = 0; event < 2; event++)
+            writer.writeEvent(String.valueOf(event)).join();
+        EventStreamReader<String> reader = clientFactory.createReader(readerGroupId + "1", readerGroupId,
+                new UTF8StringSerializer(), ReaderConfig.builder().build());
+        assertEquals(reader.readNextEvent(5000).getEvent(), "0");
+        reader.close();
+
+        // Create a Checkpoint, get StreamCut and truncate the Stream at that point.
+        StreamCut streamCut = readerGroup.getStreamCuts().get(Stream.of(scopeName, streamName));
+        assertTrue(streamManager.truncateStream(scopeName, streamName, streamCut));
+
+        // Verify that a new reader reads from event 1 onwards.
+        final String newReaderGroupName = readerGroupId + "new";
+        readerGroupManager.createReaderGroup(newReaderGroupName, ReaderGroupConfig.builder().stream(Stream.of(scopeName, streamName)).build());
+        @Cleanup
+        final EventStreamReader<String> newReader = clientFactory.createReader(newReaderGroupName + "2",
+                newReaderGroupName, new UTF8StringSerializer(), ReaderConfig.builder().build());
+
+        assertEquals("Expected read event: ", "1", newReader.readNextEvent(5000).getEvent());
+    }
+
+    /**
+    * This method is trying to check the stream seal feature of the stream.
+    * Here we are creating a stream and writing some event to it.
+    * And then sealing that stream by using stream manager and validating the stream whether sealing worked properly.
+    */
+    public  void checkSealStream() {
+        String scopeName = "stream-seal-test-scope";
+        String streamName = "stream-seal-test-stream";
+        streamManager.createScope(scopeName);
+        streamManager.createStream(scopeName, streamName, streamConfig);
+        assertTrue(streamManager.checkStreamExists(scopeName, streamName));
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scopeName, ClientConfig.builder().controllerURI(controllerURI).build());
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName, new UTF8StringSerializer(), EventWriterConfig.builder().build());
+        // Before sealing of the stream it must write to the existing stream.
+        writer.writeEvent("0").join();
+        assertTrue(streamManager.sealStream(scopeName, streamName));
+        // It must complete Exceptionally because stream is sealed already.
+        assertTrue(writer.writeEvent("1").completeExceptionally(new IllegalStateException()));
```

Comment:
You should be able to read successfully from the stream, can you add that test as well?